google/cloud/pubsub/internal/noop_message_batch.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_NOOP_MESSAGE_BATCH_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_NOOP_MESSAGE_BATCH_H
+
+#include "google/cloud/pubsub/internal/message_batch.h"
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+/**
+ * A no-op implementation of `MessageBatch`
+ */
+class NoOpMessageBatch : public MessageBatch {
+ public:
+  static std::shared_ptr<NoOpMessageBatch> Create() {
```

Comment:
A couple of last questions I'm probably just lacking context for.

Why have a factory at all, rather than just a constructor?  Normally we would resort to the former when there has to be something done outside the constructor to deliver a fully-formed object.  But that is not the case here, and seemingly could never be.

Why `shared_ptr` instead of `unique_ptr`?  Is the class overspecializing itself?

Reply:
I was using factory method because it's what I saw elsewhere.

I don't think we need it. 

Removed the `Create` and made the constructor public.

Reply:
I think that's a good choice, at least until an external requirement emerges.